pkg/controller/clusterinstallation/mattermost.go

```diff
@@ -223,75 +256,60 @@ func (r *ReconcileClusterInstallation) updateMattermostDeployment(mi *mattermost
-		job := &batchv1.Job{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      updateName,
-				Namespace: mi.GetNamespace(),
-			},
-			Spec: batchv1.JobSpec{
-				Template: corev1.PodTemplateSpec{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: map[string]string{"app": updateName},
-					},
-					Spec: *new.Spec.Template.Spec.DeepCopy(),
-				},
-			},
+		alreadyRunning, err := r.fetchRunningUpdateJob(mi, reqLogger)
+		if err != nil && k8sErrors.IsNotFound(err) {
+			reqLogger.Info("Launching update job")
+			if err = r.launchUpdateJob(mi, new, imageName, reqLogger); err != nil {
+				reqLogger.Error(err, "Launching update job failed")
+				return err
+			}
+			return errors.New("Began update job")
 		}
 
-		// Override values for job-specific behavior.
-		job.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyNever
-		for i := range job.Spec.Template.Spec.Containers {
-			job.Spec.Template.Spec.Containers[i].Command = []string{"mattermost", "version"}
+		if err != nil {
+			reqLogger.Error(err, "Error trying to determine if an update job already is running")
+			return err
 		}
 
-		err := r.client.Create(context.TODO(), job)
-		if err != nil && !k8sErrors.IsAlreadyExists(err) {
-			return err
+		if alreadyRunning.Status.CompletionTime == nil {
+			return errors.New("Update image job still running..")
 		}
+
+		// job is done, schedule cleanup
 		defer func() {
-			err = r.client.Delete(context.TODO(), job)
+			reqLogger.Info(fmt.Sprintf("Deleting job %s/%s",
+				alreadyRunning.GetNamespace(), alreadyRunning.GetName()))
+
+			err = r.client.Delete(context.TODO(), alreadyRunning)
 			if err != nil {
 				reqLogger.Error(err, "Unable to cleanup image update check job")
 			}
-		}()
 
-		// Wait up to 60 seconds for the update check job to return successfully.
-		timer := time.NewTimer(60 * time.Second)
-		defer timer.Stop()
-
-	upgradeJobCheck:
-		for {
-			select {
-			case <-timer.C:
-				return errors.New("timed out waiting for mattermost image update check job to succeed")
-			default:
-				foundJob := &batchv1.Job{}
-				err = r.client.Get(
-					context.TODO(),
-					types.NamespacedName{
-						Name:      updateName,
-						Namespace: mi.GetNamespace(),
-					},
-					foundJob,
-				)
+			podList := &corev1.PodList{}
+			listOptions := k8sClient.ListOptions{
+				LabelSelector: labels.SelectorFromSet(
+					labels.Set(map[string]string{"app": updateName})),
+				Namespace: alreadyRunning.GetNamespace(),
+			}
+
+			err = r.client.List(context.Background(), &listOptions, podList)
+			reqLogger.Info(fmt.Sprintf("Deleting %d pods", len(podList.Items)))
+			for _, p := range podList.Items {
+				reqLogger.Info(fmt.Sprintf("Deleting pod %s/%s", p.Namespace, p.Name))
+				err = r.client.Delete(context.TODO(), &p)
 				if err != nil {
-					continue
-				}
-				if foundJob.Status.Failed > 0 {
-					return errors.New("Upgrade image job check failed")
+					reqLogger.Error(err, "Problem deleting pod %s", p)
```

Comment:
Looks like we probably want a `fmt.Sprintf` here?

Reply:
Oh good catch here especially